src/svcs/_core.py

```diff
@@ -415,13 +424,13 @@ def _takes_container(factory: Callable) -> bool:
         raise TypeError(msg)
 
     ((name, p),) = tuple(sig.parameters.items())
-    if name == "svcs_container":
-        return True
-
-    if (annot := p.annotation) is Container or annot == "svcs.Container":
-        return True
+    annot = p.annotation
```

Comment:
I think we can drop this alias here – introducing a variable is probably not faster than one extra attribute lookup.